### PR TITLE
Dark light settings

### LINKED ISSUE #147: Consider using themes
I think it would be nice if we could expose a way in the UI for the user to set some pre-made themes like we do in Code.

This requires some things to solve. How do we want to store theme information? Is there an available standard format that's already doing that? If not, do we want to create one from scrach?

### PR DESCRIPTION
**Pull request by danrabbit**
Adds a setting for Dark/Light/High Contrast style Fixes #147

* Reformatted the hex_palette array and use colors from default gsettings
* Added a setting for gtk_prefer_dark and restore it on app startup
* Set prefer_dark, foreground, and background colors from the popover

screenshot from 2017-11-10 15 53 20

### CODE REVIEW

File: src/PantheonTerminalWindow.vala [hunk 1]
```diff
@@ -55,27 +55,33 @@ namespace PantheonTerminal {
...
+        public const string ACTION_CLOSE_TAB = "action_close_tab";
+        public const string ACTION_FULLSCREEN = "action_fullscreen";
+        public const string ACTION_NEW_TAB = "action_new_tab";
+        public const string ACTION_NEW_WINDOW = "action_new_window";
+        public const string ACTION_NEXT_TAB = "action_next_tab";
+        public const string ACTION_PREVIOUS_TAB = "action_previous_tab";
+        public const string ACTION_ZOOM_IN_FONT = "action_zoom_in_font";
+        public const string ACTION_ZOOM_OUT_FONT = "action_zoom_out_font";
+
+        private static Gee.MultiMap<string, string> action_accelerators = new Gee.HashMultiMap<string, string> ();
```
Comment (donadigo) [inline src/PantheonTerminalWindow.vala]: Can this be constructed in the `static construct`?

File: src/PantheonTerminalWindow.vala [hunk 2]
```diff
@@ -305,6 +316,21 @@ namespace PantheonTerminal {
             zoom_default_button.clicked.connect (() => action_zoom_default_font ());
             zoom_out_button.clicked.connect (() => action_zoom_out_font ());
 
+            color_button.mode_changed.connect (() => {
```
Comment (donadigo) [inline src/PantheonTerminalWindow.vala]: What should happen here:

The `mode_changed` signal should set `settings.prefer_dark_style`.

Then you connect to `settings.notify["prefer-dark-style"]` and change the background and foreground based on that value. (You make an `update_style` or something).

Comment (danrabbit) [reply src/PantheonTerminalWindow.vala]: In the case that we add a third style (for example a high contrast black on white style), prefering the dark stylesheet does not necessarily imply what the background and foreground colors should be.

File: src/PantheonTerminalWindow.vala [hunk 3]
```diff
@@ -305,6 +363,24 @@ namespace PantheonTerminal {
             zoom_default_button.clicked.connect (() => action_zoom_default_font ());
             zoom_out_button.clicked.connect (() => action_zoom_out_font ());
 
+            color_button_dark.clicked.connect (() => {
+                settings.prefer_dark_style = true;
+                settings.background = "rgba(37, 46, 50, 0.95)";
+                settings.foreground = "#94a3a5";
```
Comment (donadigo) [inline src/PantheonTerminalWindow.vala]: Can we get those strings as constants somewhere?

File: src/PantheonTerminalWindow.vala [hunk 4]
```diff
@@ -305,6 +363,24 @@ namespace PantheonTerminal {
...
+            });
+
+            color_button_light.clicked.connect (() => {
+                settings.prefer_dark_style = false;
+                settings.background = "rgba(253, 246, 227, 0.95)";
+                settings.foreground = "#586e75";
```
Comment (donadigo) [inline src/PantheonTerminalWindow.vala]: These too.

File: src/PantheonTerminalWindow.vala [hunk 5]
```diff
@@ -305,6 +363,24 @@ namespace PantheonTerminal {
...
+            });
+
+            color_button_white.clicked.connect (() => {
+                settings.prefer_dark_style = false;
+                settings.background = "#fff";
+                settings.foreground = "#333";
```
Comment (donadigo) [inline src/PantheonTerminalWindow.vala]: And these also.

**Pull request merged**